lib/install/reconfigure/planbuilder.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconfigure
+
+import (
+	"fmt"
+
+	"github.com/gravitational/gravity/lib/fsm"
+	"github.com/gravitational/gravity/lib/install"
+	installphases "github.com/gravitational/gravity/lib/install/phases"
+	"github.com/gravitational/gravity/lib/install/reconfigure/phases"
+	"github.com/gravitational/gravity/lib/storage"
+)
+
+// PlanBuilder builds plan for the reconfigure operation.
+type PlanBuilder struct {
+	// PlanBuilder is the embedded installer plan builder.
+	*install.PlanBuilder
+}
+
+// AddChecksPhase adds the preflight checks phase to the plan.
+func (b *PlanBuilder) AddChecksPhase(plan *storage.OperationPlan) {
+	plan.Phases = append(plan.Phases, storage.OperationPhase{
+		ID:          installphases.ChecksPhase,
+		Description: "Execute preflight checks",
+		Data: &storage.OperationPhaseData{
+			Server: &b.Master,
+		},
+	})
+}
+
+// AddPreCleanupPhase adds the phase that does pre-reconfiguration cleanups
+// on the node to the plan.
+func (b *PlanBuilder) AddPreCleanupPhase(plan *storage.OperationPlan) {
+	plan.Phases = append(plan.Phases, storage.OperationPhase{
+		ID:          phases.PreCleanupPhase,
```

Comment:
`PreCleanup` does not relay too much meaning. As far as I understand, the split is between things running prior to kubernetes being available and after  - what about splitting around kubernetes? Something like `PreKubernetes` and `PostKubernetes`?